feat(dashboard compare): Save mutation testing result in stryker dashboard to re-use in later run (#1067)
stryker-mutator/stryker-net: docs/Configuration.md

```

Default: `master`

## Dashboard Compare

> Enabling the dashboard compare feature saves reports and re-uses the result when a mutant or it's tests are unchanged.

Reviewer: ```suggestion
Dashboard compare lets you save your mutation result in [the stryker dashboard](https://dashboard.stryker-mutator.io). On subsequent test runs only changed mutants, or mutants for which a covering unit test has been changed, will be tested. Unchanged mutants will get the status from the dashboard in order to provide a full report.
```

```
dotnet stryker --dashboard-compare
dotnet stryker -compare
```

Default `"off"`

This feature automatically enables the --diff feature.

## Fallback version

> When enabling the --dashboard-compare feature you can provide a fallback version. This version will be used to pull a baseline when we cannot find a baseline for your current branch. When we are still unable to provide a baseline we will start a complete testrun to create a complete baseline.

Reviewer: ```suggestion
When enabling the --dashboard-compare feature you can provide a fallback version. This version will be used to download a report when we cannot find one for your current source version. When we are unable to find a fallback version we will do a complete mutation testrun.
```

```
dotnet stryker --dashboard-compare --dashboard-fallback-version master
dotnet stryker -compare -fallback-version master
```
Default: `null`

## Configuring Dashboard Compare on pull requests
When configuring the --dashboard-compare feature on pull requests please provide the following configurations.

> 1.  Enable --dashboard-compare.

Reviewer: ```suggestion
1. Enable --dashboard-compare.
```

2. Set --dashboard-version to the name of the source branch for your pull request.

> 3. Set --dashboard-fallback-version to the name of the target branch for your pull request.

Reviewer: We should allow for only a fallback version or only a git source to be set. If only one is set, it should be used for both. If we recommend to use the same value for both, we should handle that case as painless as possible.

Author: Right now GitSource is not allowed to be null, this will result in an exception. I have added a check in StrykerOptions that checks if FallbackVersion is null. If this is the case it is set to GitSource.

4. Set --git-source to the name of the target branch of your pull request.

```

> dotnet stryker --dashboard-compare --dashboard-fallback-version master -git-source master --dashboard-version development

Reviewer: ```suggestion
dotnet stryker --dashboard-compare --dashboard-fallback-version master --git-source master --dashboard-version development
```

> dotenet stryker -compare -fallback-version master -source master -version development

Reviewer: ```suggestion
dotnet stryker -compare -fallback-version master -source master -version development
```
